docs/section5/Rule5-38.md

```diff
@@ -0,0 +1,89 @@
+
+- Get skylight-roof-ratio for semi-exterior type roofs: `srr_semi_exterior = rmr_scc_skylight_roof_ratios_dictionary["SEMI-EXTERIOR"]`
+
+  - If skylight-roof-ratio is greater than 2.0%, get baseline skylight construction requirement: `if srr_semi_exterior > 0.02: target_shgc_semiheated = data_lookup(table_G3_4, climate_zone, "SEMIHEATED, "SKYLIGHT", "2.1%+", "ASSEMBLY MAX. SHGC")`
+
+  - Else, skylight-roof-ratio is 0% to 2.0%, get baseline skylight construction requirement: `else: target_shgc_semiheated = data_lookup(table_G3_4, climate_zone, "SEMIHEATED", "SKYLIGHT", "0%-2.0%", "ASSEMBLY MAX. SHGC")`
+
+- For each zone in B_RMR: `for zone_b in B_RMR...zones:`
+
+  - For each surface in zone: `for surface_b in zone_b.surfaces:`
+
+    - Check if surface is roof with subsurface and is regulated: `if ( get_opaque_surface_type(surface_b) == "ROOF"  ) AND ( surface_b.subsurfaces ) AND ( scc_dictionary_b[surface_b.id] != "UNREGULATED" ):`
+
+      - For each subsurface in roof: `for subsurface_b in surface_b.subsurfaces:`
+
+        - Check if subsurface is door and glazed area is more than 50% of the total door area, or subsurface is not door: `if (( subsurface_b.classification == "DOOR" ) AND ( subsurface_b.glazed_area > subsurface_b.opaque_area )) OR ( subsurface_b.classification != "DOOR" ):`
+
+          **Rule Assertion - Component:**
+
+          - Case 1； For each subsurface, if roof is exterior mixed type and the baseline requirements for residential and non-residential type SHGC for different skylight-roof-ratio are different: `if manual_review_flag AND ( scc_dictionary_b[surface_b] == "EXTERIOR MIXED" ): UNDETERMINED`
+
+          - Case 2: Else if roof is exterior mixed type and skylight SHGC matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR MIXED" ) AND ( subsurface_b.solar_heat_gain_coefficient == target_shgc_mixed ): PASS`
+
+          - Case 3: Else if roof is exterior residential type and skylight SHGC matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR RESIDENTIAL" ) AND ( subsurface_b.solar_heat_gain_coefficient == target_shgc_res ): PASS`
+
+          - Case 4: Else if roof is exterior non-residential type and skylight SHGC matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "EXTERIOR NON-RESIDENTIAL" ) AND ( subsurface_b.solar_heat_gain_coefficient == target_shgc_nonres ): PASS`
+
+          - Case 5: Else if roof is semi-exterior type and skylight SHGC matches Table G3.4 requirement: `else if ( scc_dictionary_b[surface_b] == "SEMI-EXTERIOR" ) AND ( subsurface_b.solar_heat_gain_coefficient == target_shgc_semiheated ): PASS`
+
+          - Case 6: Else: `else: FAIL and NUMBER_OF_FAIL_COMPONENTS++`
+
+**Rule Assertion - RMR:**
+
+- Case 1: If any subsurface in B-RMR is ruled as "UNDETERMINED": `UNDETERMINED and raise_message "MANUAL REVIEW IS REQUESTED TO VERIFY SKYLIGHT MEETS SHGC REQUIREMENT AS PER TABLE G3.4."`
```

Comment:
I don't think we should require manual review of all surfaces because there is one surface that requires a manual review